### PR TITLE
CXXCBC-216: allow to use movable_function for replica callbacks

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

File: core/operations/document_get_all_replicas.hxx [hunk 1]
```diff
@@ -58,7 +58,7 @@ struct get_all_replicas_request {
               if (ec) {
                   return h(response_type{ make_key_value_error_context(ec, id) });
               }
-              using handler_type = std::function<void(response_type)>;
+              using handler_type = utils::movable_function<void(response_type)>;
```
Comment (thejcfactor) [inline core/operations/document_get_all_replicas.hxx:61]: Is there a missing include (i.e. #include "core/utils/movable_function.hxx")?

File: core/operations/document_get_any_replica.hxx [hunk 2]
```diff
@@ -55,7 +55,8 @@ struct get_any_replica_request {
               if (ec) {
                   return h(response_type{ make_key_value_error_context(ec, id) });
               }
-              using handler_type = std::function<void(response_type)>;
+              using handler_type = utils::movable_function<void(response_type)>;
```
Comment (thejcfactor) [inline core/operations/document_get_any_replica.hxx:58]: Is there a missing include (i.e. #include "core/utils/movable_function.hxx")?

**Review by thejcfactor — COMMENTED** [review]
LGTM -- pending possible missing header includes.

File: core/operations/document_get_any_replica.hxx (hunk 2 above)
Comment (avsej) [reply core/operations/document_get_any_replica.hxx:58]: it is probably, because of test include everything it didn't fail here. I will fix it.

**Review by thejcfactor — APPROVED** [review]
LGTM 👍

**Pull request merged**